typescript/sdk/src/token/deploy.ts

```diff
@@ -65,6 +59,51 @@ export class HypERC20Deployer extends GasRouterDeployer<
     }); // factories not used in deploy
   }
 
+  routerContractName<K extends keyof HypERC20Factories>(
+    config: ERC20RouterConfig,
+  ): K {
+    if (isCollateralConfig(config)) {
+      return (
+        isFastConfig(config) ? TokenType.fastCollateral : TokenType.collateral
+      ) as K;
+    } else if (isSyntheticConfig(config)) {
+      return (
+        isFastConfig(config) ? TokenType.fastSynthetic : TokenType.synthetic
+      ) as K;
+    } else {
+      throw new Error('Unknown collateral type when constructing router name');
+    }
+  }
```

Comment:
oh I see, interesting

Reply:
I think @paulbalaji also changed this in https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/3490

Reply:
I think Paul's PR will compliment this. I've noted to merge his first so i can rebase